src/it/java/com/google/checkstyle/test/base/AbstractGoogleModuleTestSupport.java

```diff
@@ -92,6 +92,17 @@ protected static Configuration getModuleConfig(String moduleName) {
         return getModuleConfig(moduleName, null);
     }
 
+    /**
+     * Returns {@link Configuration} instance for the given module configuration.
+     * This implementation uses {@link #getModuleConfig(String, String)} method inside.
+     *
+     * @param moduleConfig module configuration.
+     * @return {@link Configuration} instance for the given module configuration.
+     */
+    protected static Configuration getModuleConfig(DefaultConfiguration moduleConfig) {
+        return moduleConfig;
```

Comment:
This method makes no sense. The method is static and taking the parameter and just returning as is. `DefaultConfiguration` is already castable to `Configuration` showcase by the effect that it just returns the parameter without doing anything else.

Reply:
@rnveach Yeah ,basically this method is not serving anything as such. But had to add this because of [CheckstyleTestMakeupCheck](https://sevntu-checkstyle.github.io/sevntu.checkstyle/apidocs/com/github/sevntu/checkstyle/checks/design/CheckstyleTestMakeupCheck.html).It is raising violation.

Reply:
please remove this method.

Reply:
@romani I tried to remove  it but getting violation from seventu for CheckstyleTestMakeupCheck.

Reply:
please remove it to let me see a failure, it is not clear why it is failing, if that is false positive, we will show how to suppress it.

Reply:
Yup, doing it now

Reply:
![image](https://user-images.githubusercontent.com/42171790/114563426-06014380-9c8d-11eb-8669-2fe6e0412854.png)
Getting this error @romani.

I have pushed the changes.

Reply:
This error is coming from https://github.com/sevntu-checkstyle/sevntu.checkstyle/blob/master/sevntu-checks/src/main/java/com/github/sevntu/checkstyle/checks/design/CheckstyleTestMakeupCheck.java#L231 and it is controlled by the property `createMethodRegexp` which defaults to `create(Root|Module)Config|getModuleConfig`.

Reply:
The main reason why `createTreeWalkerConfig` was not originally allowed was because of the fact that the normal `verify` method has built in code to identify if it is related to `TreeWalker` or not. It is redundant to specifically call it in normal circumstances.

The check follows the same restrictions as Checkstyle in that we can't identify method parameters and such.

@romani It depends on how much we want to build into this new functionality. Should the existing `verify` method continue to be smart and identify that the config being given falls under `TreeWalker`?

I want to avoid allowing all uses of `createTreeWalkerConfig`, but I don't see us requiring a comment suppression to ignore the violation. This check cleaned up some shananigans when it was first introduced/

Reply:
Can we  create `getModuleConfig(DefaultConfiguration config)` as I did previously which will  just return  the object sent to it. Although its a very trivial one, but solving the issue

Reply:
I do not see much problems to allow a bit more smart test code to, at least we know what is reason for this, and reason is very good - we need to test how config(few Checks) works, not a specific Check.